src/network/netmon.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include "sntp_client.h"
+#include "dhcpwait.h"
+#include "netmon_stats.h"
+
+LOG_MODULE_REGISTER(netmon, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define NETMON_THREAD_PRIORITY 5
+static struct k_work_delayable time_sync;
+
+/* Network monitor worker thread work item to sync
+ * the time with SNTP server.
+ */
+static void netmon_time_sync(struct k_work *work)
+{
+
+	if (!is_dhcp_up()) {
+		LOG_ERR("Network down");
+		goto end;
```

Comment:
I think this works, but let me make sure things are clear. We re-schedule the timer for 24 hours when the network is down. But the monitoring thread will also notice the network is down, and cancel this timer, and schedule a new one for a shorter interval?